core/aws-core/src/main/java/software/amazon/awssdk/awscore/exception/AwsServiceException.java

```diff
@@ -67,9 +73,28 @@ public String getMessage() {
 
     @Override
     public boolean isClockSkewException() {
-        return Optional.ofNullable(awsErrorDetails)
-                .map(a -> AwsErrorCode.CLOCK_SKEW_ERROR_CODES.contains(a.errorCode()))
-                .orElse(false);
+        if (super.isClockSkewException()) {
+            return true;
+        }
+
+        if (awsErrorDetails == null) {
+            return false;
+        }
+
+        if (AwsErrorCode.DEFINITE_CLOCK_SKEW_ERROR_CODES.contains(awsErrorDetails.errorCode())) {
+            return true;
+        }
+
+        SdkHttpResponse sdkHttpResponse = awsErrorDetails.sdkHttpResponse();
+
+        if (clockSkew == null || sdkHttpResponse == null) {
+            return false;
+        }
+
+        return (AwsErrorCode.POSSIBLE_CLOCK_SKEW_ERROR_CODES.contains(awsErrorDetails.errorCode()) ||
```

Comment:
Can we break down the conditions so that it's easier to read?

Reply:
I was on the border about making `AwsErrorCode.isPossibleClockSkewErrorCode(...)`, so I think you've convinced me with this question.